### DISCUSSION TITLE
Struggling with a type error when transforming

### DISCUSSION THREAD

**Discussion by johtso** [colinhacks/zod#2145] [Q&A] [answered]
Any ideas what I'm doing wrong here? [typescript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAVygGzgMyhEcDkTk4BQokscA3nAF4A0cAWhACYAqAnmAKYCCAdm3AC+6TNhxVmREtHiUAxlE4BDGJ0ZMAYpxhyAFpyhCRWXBKYBaNNr1FCAejtxkEAOZwY+uFCUB3L5wBnSF4AzjgmFSV3CDh9ZDA4H2APcM4AIwQXF2BeF0I5CBD4Kx1dAHVk3XU4AF44BWVVdS1SgwAKNvw6HOSASlqAPgpCOH8YJF50a11OlG7ePoA6D05eNsUAweHR0cVxqEmNxYArAMK23uX9NdPCrfIRndGCkIhkTkXnFzbb3l7Hp57CZwX4AoT-UaCf5QwiEGAcMIAUV4TEgORgADklCAwnUAEShJRQPQAYQQARgWAMeLgAB84HiAG5KZDACKqAAKSgCAR80CYeIA3LDuWxeHJ0AhxTBgHcAiR3s1pgAeFgAZT0nBAUU4AA9VCjNup2Fw+GwBm1VqiIOiAFxwZHW9FYnF0AKa7X29UepR0MBE7EBe0AJU4BSgTGVFKgORcdCU-AG-Qez0KFLgAEdNnVeJw-PhFgBVYMAGTVymJui53hAATa-prAX+jxe6f0SiYBmz2x2OBeBpg5nhXBw9pwSjAYFZchUst4dl15h8y8s0BA5nwqwKnaYOBoj0EsNTRWoGgQyGQoaCadx1DBiwgaWOYZgbRTTzgZxxACFuZwANKcGw9ruvo2r7k8MJPMs3ghGga5tOyUQ1EM76AtowJIYsX6cL+oSAWwYKjA4H4AHoAPxgjCjxAgcUylBUHjqG0VBnheV7BKEdB4roMAwGAQYOHq2JTh8BQgHidBoXAOIeMwY4cgA8mqLB7mC7adlAQaxMomkBBBOxpMwwGZgEywQGqMAxrkFwGTCh6EHqpDwKK4qStKc5wMyrLspwXI8nyEb1gGtb2pQWpKMAyAgVZsZ+tyvL8jF1luFCPa0ZM8rgIqzAtHobRgjg3lsiofkJYFu4GaMVAPk+L5vkR9TkpSOJQCSzCcPaNXRrGtlUb0VVwA2gaPNChBAA)

Is `TSchema extends ZodTypeAny` maybe not the right thing to do?

**Comment by JacobWeisenburger** [comment, marked as answer]
duplicate of https://github.com/colinhacks/zod/issues/2146
